src/ray/gcs/gcs_server/test/gcs_placement_group_manager_test.cc

```diff
@@ -128,6 +130,25 @@ TEST_F(GcsPlacementGroupManagerTest, TestGetPlacementGroupIDByName) {
           create_placement_group_request.placement_group_spec().placement_group_id()));
 }
 
+TEST_F(GcsPlacementGroupManagerTest, TestRescheduleWhenNodeAdd) {
+  auto create_placement_group_request = Mocker::GenCreatePlacementGroupRequest();
+  std::atomic<int> finished_placement_group_count(0);
+  gcs_placement_group_manager_->RegisterPlacementGroup(
+      create_placement_group_request,
+      [&finished_placement_group_count]() { ++finished_placement_group_count; });
+  ASSERT_EQ(finished_placement_group_count, 0);
+  ASSERT_EQ(mock_placement_group_scheduler_->placement_groups.size(), 1);
+  auto placement_group = mock_placement_group_scheduler_->placement_groups.back();
+  mock_placement_group_scheduler_->placement_groups.pop_back();
+
+  // If the creation of placement group fails, it will be rescheduled after a short time.
```

Comment:
This doesn't seem to have any simulation to add nodes?

Reply:
We can't test here because it needs to start gcs server and if a node is added, `SchedulePendingPlacementGroups ` will be called directly, so we test reschedule here.